kucher/view/main_window/telega_control_widget/task_specific_status_widget/fault_status_widget.py

```diff
@@ -83,7 +84,16 @@ def on_general_status_update(self, timestamp: float, s: GeneralStatusView):
         self._error_code_hex.setText(f'0x{tssr.failed_task_exit_code:02X}')
         self._error_code_bin.setText(f'0b{tssr.failed_task_exit_code:08b}')
 
-        self._error_description_display.setText('(elaboration not available)')
+        file_name = 'kucher/view/main_window/telega_control_widget/task_specific_status_widget/error_codes.yml'
+        with open(file_name, 'r') as f:
+            error_codes = yaml.safe_load(f)
+
+        error_description = 'unknown error'
+        for c, d in error_codes[str(tssr.failed_task_id).split('.')[-1]].items():
+            if c == tssr.failed_task_exit_code:
+                error_description = d
```

Comment:
This code gets the job done but you should be aware that the `dict` class provides a method `get()` which can default to a predefined value if the key is not found:

```python
failed_task_name = str(tssr.failed_task_id).split('.')[-1]
error_description = error_codes[failed_task_name].get(tssr.failed_task_exit_code, 'unknown error')
```